lib/RVSDG/RVSDGOps.cpp

```diff
@@ -100,53 +107,57 @@ LogicalResult GammaResult::verify() {
-      return emitOpError("Mismatched types between lambda signature and body "
-                         "arguments. ")
-             << "Signature argument #" << sigI << " has type "
-             << signatureParamTypes[sigI] << ". Body argument #" << argI
-             << " has type " << bodyArgumentTypes[argI];
-    }
-  }
-
-  for (size_t inI = 0, argI=signatureParamTypes.size(); inI < nodeInputTypes.size(); ++inI, ++argI) {
-    if (nodeInputTypes[inI] != bodyArgumentTypes[argI]) {
-      return emitOpError("Mismatched types in lambda node body arguments. "
-                         "First arguments "
-                         "should match node inputs. "
-                         "Offending argument: #")
-             << argI << " Expected " << nodeInputTypes[inI] << ", got "
-             << bodyArgumentTypes[argI];
-    }
-  }
+  // Value signatureVal = this->getResult();
+  // if (!signatureVal.getType().isa<LambdaRefType>()) {
+  //   return emitOpError(
+  //       "Result type is invalid. Expected an instance of LambdaRefType.");
+  // }
+  // LambdaRefType signatureType = signatureVal.getType().cast<LambdaRefType>();
+
+  // ArrayRef<Type> signatureParamTypes = signatureType.getParameterTypes();
+  // auto bodyArgumentTypes = this->getRegion().getArgumentTypes();
+  // auto nodeInputTypes = this->getOperandTypes();
+
+  // if (bodyArgumentTypes.size() != signatureParamTypes.size() + nodeInputTypes.size()) {
+  //   return emitOpError("Mismatched number of arguments between lambda signature and body. ")
+  //          << "Signature has " << signatureParamTypes.size() << " arguments\n"
+  //          << "Node has " << nodeInputTypes.size() << " inputs\n"
+  //          << "Total number of arguments should be " << signatureParamTypes.size() +
+  //          nodeInputTypes.size() << "\n"
+  //          << "Body has " << bodyArgumentTypes.size() << " arguments. ";
+  // }
+
+  // for (size_t sigI = 0, argI = 0;
+  //      sigI < signatureParamTypes.size(); ++sigI, ++argI) {
+  //   if (signatureParamTypes[sigI] != bodyArgumentTypes[argI]) {
+  //     return emitOpError("Mismatched types between lambda signature and body "
+  //                        "arguments. ")
+  //            << "Signature argument #" << sigI << " has type "
+  //            << signatureParamTypes[sigI] << ". Body argument #" << argI
+  //            << " has type " << bodyArgumentTypes[argI];
+  //   }
+  // }
+
+  // for (size_t inI = 0, argI=signatureParamTypes.size(); inI < nodeInputTypes.size(); ++inI,
+  // ++argI) {
+  //   if (nodeInputTypes[inI] != bodyArgumentTypes[argI]) {
+  //     return emitOpError("Mismatched types in lambda node body arguments. "
+  //                        "First arguments "
+  //                        "should match node inputs. "
+  //                        "Offending argument: #")
+  //            << argI << " Expected " << nodeInputTypes[inI] << ", got "
+  //            << bodyArgumentTypes[argI];
+  //   }
+  // }
```

Comment:
Why is validation of LambdaNode commented out?

Reply:
If I understand correctly, this only validates that the lambdaRefType (which was previously used as a non-opaque pointer to a lambda) matches the types of inputs and outputs of the function represented by the lambda. Removing the double-encoding of this type information removes the need for validation.